src/App.jsx

```diff
@@ -14,45 +15,106 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_BY_ALPHABET = 'alphabet';
+const SORT_BY_LENGTH = 'length';
+const SORT_REVERSED = 'reversed';
+
+export const App = () => {
+  const [goods, setGoods] = useState(goodsFromServer);
+  const [sortBy, setSortBy] = useState('');
+  const [reversed, setReversed] = useState(false);
+
+  const sortGoods = (type) => {
+    if (type === SORT_BY_LENGTH) {
+      setSortBy(SORT_BY_LENGTH);
+      const sortedGoods = [...goodsFromServer].sort(
+        (g1, g2) => g1.length - g2.length,
+      );
+
+      if (reversed) {
+        const reverseSortedGoods = sortedGoods.reverse();
+
+        setGoods(reverseSortedGoods);
+      } else {
+        setSortBy(SORT_BY_LENGTH);
+        setGoods(sortedGoods);
+      }
+    } else if (type === SORT_BY_ALPHABET) {
+      setSortBy(SORT_BY_ALPHABET);
+      if (reversed) {
+        setGoods([...goods].sort().reverse());
+      } else {
+        setGoods([...goods].sort());
+      }
+    } else if (type === SORT_REVERSED) {
+      setGoods(
+        [...goods].reverse(),
+      );
+    }
```

Comment:
It looks a bit complicated and confusing, you need to use a structure switch/case